apax/train/checkpoints.py

```diff
@@ -14,13 +14,56 @@
 log = logging.getLogger(__name__)
 
 
-def load_state(model, params, tx, ckpt_dir):
+def check_for_ensemble(params):
+    flat_params = flatten_dict(params)
+    shapes = [v.shape[0] for v in flat_params.values()]
+    is_ensemble = shapes == shapes[::-1]
+
+    if is_ensemble:
+        return shapes[0]
+    else:
+        return 1
+
+
+def create_train_state(model, params, tx):
+    n_models = check_for_ensemble(params)
+
+    def inner(params):
+        state = train_state.TrainState.create(
+            apply_fn=model,
+            params=params,
+            tx=tx,
+        )
+        return state
```

Comment:
I don't like the function name, could it be `compute_inner` ? `inner` sounds more like a variable.